src/main/java/com/example/pfairplayservice/controller/MemberController.java

```diff
@@ -51,13 +52,24 @@
     }
 
     @PutMapping("/member/{uid}")
-    public ResponseEntity<Void> updatePasswordByUid(@PathVariable String uid, @RequestParam String password) {
+    public ResponseEntity<Void> updatePasswordByUid(@PathVariable String uid, @RequestBody Member updateMember) {
         Optional<MemberEntity> member = memberRepository.findById(uid);
         if (!member.isPresent()) {
             throw new SourceNotFoundException(String.format("uid{%s} not found", uid));
         }
-        member.get().setPassword(password);
-        memberRepository.save(member.get());
+
+        if (member.get().getPassword() != updateMember.getPassword())
+            memberRepository.updatePasswordByUid(uid, updateMember.getPassword());
+        if (member.get().getAddress() != updateMember.getAddress())
+            memberRepository.updateAddressByUid(uid, updateMember.getAddress());
+        if (member.get().getPhoneNumber() != updateMember.getAddress())
+            memberRepository.updatePhoneNumberByUid(uid, updateMember.getPhoneNumber());
+        if (member.get().getPreferPosition() != Position.to(updateMember.getPreferPosition()))
+            memberRepository.updatePreferPositionByUid(uid, Position.to(updateMember.getPreferPosition()));
```

Comment:
FIx `Position.to(updateMember.getPreferPosition())` -> `updateMember.getPreferPosition().getPosition()`

Reply:
I will.